src/backend/distributed/utils/resource_lock.c

```diff
@@ -180,23 +205,44 @@ lock_shard_resources(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errmsg("no locks specified")));
 	}
 
+	if (lockMode == AccessExclusiveLock)
+	{
+		/* executor does not acquire this level, and prevent any malicious user */
+		ereport(ERROR, (errmsg("cannot acquire AccessExclusiveLock")));
```

Comment:
This check is not covered

Reply:
Please see the other comment, they are related anyway.

Reply:
As said in my other comment, let's remove this check.